usaspending_api/database_scripts/etl/award_delta_view.sql

```diff
@@ -71,7 +71,7 @@ SELECT
       THEN CONCAT(
         '{"name":"', vw_award_search.funding_toptier_agency_name,
         '","code":"', vw_award_search.funding_toptier_agency_code,
-        '","id":"', vw_award_search.funding_agency_id, '"}'
+        '","id":"', (SELECT aa.id FROM Agency aa WHERE aa.toptier_agency_id = (SELECT aaa.toptier_agency_id FROM Agency aaa WHERE aaa.id = vw_award_search.funding_agency_id) AND aa.toptier_flag = True), '"}'
```

Comment:
Wouldn't it make more sense to select the toptier_agency_id instead of the agency id? I'm pretty sure that this will still group by the agency table instead of the toptier agency table.

Reply:
The agency IDs need to be the PK from the agency table. The rub is they need to be the _same_ id for the records

Reply:
I see, so the strategy here is to filter on the toptier flag, which should only be true once per toptier ID. That looks consistent to me.

Reply:
Yeah, it appears to work well enough for the postgres api responses. There are a few edgecases so it isn't perfect.

Reply:
Improved SQL. Thanks @kbard